File: scaffold/src/main/java/top/xuqingquan/web/system/WebListenerManager.java
package top.xuqingquan.web.system;

import android.webkit.DownloadListener;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public interface WebListenerManager {

    @SuppressWarnings("UnusedReturnValue")
    WebListenerManager setWebChromeClient(WebView webview, WebChromeClient webChromeClient);

    @SuppressWarnings("UnusedReturnValue")
    WebListenerManager setWebViewClient(WebView webView, WebViewClient webViewClient);

    WebListenerManager setDownloader(WebView webView, DownloadListener downloadListener);
}

File: scaffold/src/main/java/top/xuqingquan/web/system/WebListenerManager.kt
package top.xuqingquan.web.system

import android.webkit.DownloadListener
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient

interface WebListenerManager {

    fun setWebChromeClient(webView: WebView?, webChromeClient: WebChromeClient): WebListenerManager

    fun setWebViewClient(webView: WebView?, webViewClient: WebViewClient): WebListenerManager

    fun setDownloader(webView: WebView?, downloadListener: DownloadListener?): WebListenerManager
}